site/deploy-site.js

```diff
@@ -0,0 +1,52 @@
+const { execSync } = require('child_process');
+const process = require('process');
+const path = require('path');
+const args = require('minimist')(process.argv.slice(2));
+
+const execWithOutput = (cmd) => execSync(cmd, { stdio: 'inherit' });
+
+const DEPLOY_REPOSITORY = 'https://github.com/embark-framework/embark-site';
+const DEPLOY_REMOTE = 'embark-site';
+
+const TMP_DEPLOY_BRANCH = 'embark-site-deploy';
+const LOCAL_DEPLOY_BRANCH = 'embark-site-deploy-dist';
+const REMOTE_DEPLOY_BRANCH = 'gh-pages';
+const MASTER_BRANCH = 'master';
+
+const SITE_DIR = 'site';
+const PUBLIC_DIR = 'public';
+const ROOT_DIR = path.join(__dirname, '..');
+
+const remote = args['deploy-remote'] || DEPLOY_REMOTE;
+const repository = args['deploy-repository'] || DEPLOY_REPOSITORY;
```

Comment:
It's awesome to have this be configurable, great work!

But I don't understand the strict need for `DEPLOY_REPOSITORY`. Imagine I want to use `my-fake-embark-site` remote. When I setup that remote, I will have pointed it to something like `/Users/michael/temp/my-fake-embark-site`, i.e. a path to another repo on my local filesystem.

So then for `deploy-site` script to work, I need to supply that same URL on the command-line, and in practice I'll either recall it from memory or copy/paste from my git config.

What if we checked whether `remote === DEPLOY_REMOTE`, i.e. it's `embark-site` (the default), and only in that case check that remote's URL `=== DEPLOY_REPOSITORY`?

That would catch a badly configured `embark-site` remote for real deployments and not be a hassle when doing test deployments with `--deploy-remote`.

Reply:
Sorry, you're right, this setting doesn't make a lot of sense anymore when making those things configurable via passing down a remote. Will update.